lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -736,36 +769,43 @@ createWallet ctx wid a0 a1 =
     re = ctx ^. workerRegistry @s @t @k
     df = ctx ^. dbFactory @s @t @k
 
--- Populate an API transaction record with 'TransactionInfo' from the wallet
--- layer.
-mkApiTransactionFromInfo :: TransactionInfo -> ApiTransaction t
-mkApiTransactionFromInfo (TransactionInfo txid ins outs meta depth txtime) =
-    apiTx { depth, insertedAt }
-  where
-    apiTx = mkApiTransaction txid ins outs meta
-    insertedAt = Just (ApiBlockData txtime (ApiT (meta ^. #slotId)))
-
 mkApiTransaction
     :: forall t.
        Hash "Tx"
     -> [(TxIn, Maybe TxOut)]
     -> [TxOut]
-    -> W.TxMeta
+    -> (W.TxMeta, UTCTime)
+    -> Lens' (ApiTransaction t) (Maybe ApiTimeReference)
     -> ApiTransaction t
-mkApiTransaction txid ins outs meta = ApiTransaction
-    { id = ApiT txid
-    , amount = meta ^. #amount
-    , insertedAt = Nothing
-    , depth = Quantity 0
-    , direction = ApiT (meta ^. #direction)
-    , inputs = [ApiTxInput (fmap convertTxOut o) (ApiT i) | (i, o) <- ins]
-    , outputs = NE.fromList (convertTxOut <$> outs)
-    , status = ApiT (meta ^. #status)
-    }
+mkApiTransaction txid ins outs (meta, timestamp) setTimeReference =
+    tx & setTimeReference .~ Just timeReference
   where
-      convertTxOut :: TxOut -> AddressAmount t
-      convertTxOut (TxOut addr (Coin c)) =
-          AddressAmount (ApiT addr, Proxy @t) (Quantity $ fromIntegral c)
+    tx :: ApiTransaction t
+    tx = ApiTransaction
+        { id = ApiT txid
+        , amount = meta ^. #amount
+        , insertedAt = Nothing
+        , pendingSince = Nothing
+        , depth = Quantity 0
+        , direction = ApiT (meta ^. #direction)
+        , inputs = [ApiTxInput (fmap convertTxOut o) (ApiT i) | (i, o) <- ins]
+        , outputs = NE.fromList (convertTxOut <$> outs)
+        , status = ApiT (meta ^. #status)
+        }
+
+    timeReference :: ApiTimeReference
+    timeReference = ApiTimeReference
+        { time = timestamp
+        , block = ApiBlockReference
+            { slotNumber = meta ^. (#slotId . #slotNumber)
+            , epochNumber = meta ^. (#slotId . #epochNumber)
+            }
+        }
+
+    convertTxOut :: TxOut -> AddressAmount t
+    convertTxOut (TxOut addr (Coin c)) =
```

Comment:
Perhaps `addressAmountFromTxOut` would be a clearer name here?

Reply:
`toAddressAmount` would be sufficient in this context I think. Although, that function was already there ^.^